src/openforms/api/tests/test_validators.py

```diff
@@ -0,0 +1,76 @@
+from unittest import TestCase
+
+from django.core.files.uploadedfile import SimpleUploadedFile
+from django.utils.translation import activate, get_language
+
+from rest_framework.exceptions import ValidationError
+
+from .. import validators
+
+PIXEL_GIF = b"GIF89a\x01\x00\x01\x00\x81\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00!\xf9\x04\x01\x00\x00\x00\x00,\x00\x00\x00\x00\x01\x00\x01\x00\x00\x08\x04\x00\x01\x04\x04\x00;"
+PIXEL_PNG = b"\x89PNG\r\n\x1a\n\x00\x00\x00\rIHDR\x00\x00\x00\x01\x00\x00\x00\x01\x08\x02\x00\x00\x00\x90wS\xde\x00\x00\x00\x0cIDATx\x9cc```\x00\x00\x00\x04\x00\x01\xf6\x178U\x00\x00\x00\x00IEND\xaeB`\x82"
+
+
+class MimeTypeValidatorTests(TestCase):
+    CORRECT_GIF = SimpleUploadedFile("pixel.gif", PIXEL_GIF, content_type="image/gif")
+    _original_language = get_language()
+
+    def setUp(self):
+        self.CORRECT_GIF.seek(0)
+
+    def setUpClass():
+        activate("en")
+
+    def tearDownClass():
+        activate(MimeTypeValidatorTests._original_language)
```

Comment:
`override_settings` only works with subclasses of `django.test.SimpleTestCase` this is a plain vanilla `unittest.TestCase` because it doesn't hit any of the django magic, it unittests a function.
Changing to a `SimpleTestCase` is more than 2 orders slower. At the rate I run tests during development, that counts.
```
........                                                                 [100%]
8 passed in 0.08s

==================================== ERRORS ====================================
_________ ERROR collecting src/openforms/api/tests/test_validators.py __________
test_validators.py:16: in <module>
    class MimeTypeValidatorTests(TestCase):
../../../../env/lib/python3.8/site-packages/django/test/utils.py:392: in __call__
    return self.decorate_class(decorated)
../../../../env/lib/python3.8/site-packages/django/test/utils.py:469: in decorate_class
    raise ValueError(
E   ValueError: Only subclasses of Django SimpleTestCase can be decorated with override_settings
=========================== short test summary info ============================
ERROR test_validators.py - ValueError: Only subclasses of Django SimpleTestCa...
!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
1 error in 0.27s

[1]+  Stopped                 vi test_validators.py
(env) chars@Macaca:~/Projects/open-forms/src/openforms/api/tests$ fg
vi test_validators.py
........                                                                 [100%]
8 passed in 10.15s
```

Reply:
that's odd - I can't imagine where that time is spent since `SimpleTestCase` isn't even allowed to touch the database.

Are you running tests locally with `--keepdb`? You can also tweak your local postgres instance to not flush to disk (I have set up a second cluster just for running tests locally on port 5433) to keep things fast.

We don't really use bare `unittest.TestCase` anywhere exactly because we're always operating in the Django context and expect to be able to use the django testing tools.

Also running it locally to see the difference, you may be onto something with the magic DB being different:

```
======================================================================
ERROR: test_accepts_unknown_extensions (openforms.api.tests.test_validators.MimeTypeValidatorTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/bbt/code/open-forms/src/openforms/api/tests/test_validators.py", line 54, in test_accepts_unknown_extensions
    self.assertIs(validators.mime_type_validator(file), file)
  File "/home/bbt/code/open-forms/src/openforms/api/validators.py", line 88, in mime_type_validator
    return validate_file(arg)
  File "/home/bbt/code/open-forms/src/openforms/api/validators.py", line 68, in validate_file
    raise serializers.ValidationError(
rest_framework.exceptions.ValidationError: [ErrorDetail(string="The file 'pixel.gif' is not a .gif.", code='invalid')]

----------------------------------------------------------------------
Ran 8 tests in 0.019s
```

Timings:

* `unittest.TestCase`: 0.019s
* `django.test.SimpleTestCase` without `--keepdb`: 0.004s
* `django.test.SimpleTestCase` with `--keepdb`: 0.004s

So I'm not seeing that two orders of magnitude slowdown :thinking:

Reply:
Good catch on coupling to the translations! This removes the need for the class setup and teardown:

```python
@patch('openforms.api.validators._', lambda s: s)  # decouple from translations by returning msgid
class MimeTypeValidatorTests(TestCase):
```